### PR TITLE
puppet-bolt: 3.19.0 → 3.24.0

### PR DESCRIPTION
**Pull request by uvNikita**
###### Description of changes

Updated package to 3.24.0
Also added:
 - meta info
 - simple version test
 - `passthru.updateScript`
 - prevent shebang patching

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/puppet/puppet-bolt/default.nix [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
     # Set BOLT_GEM=1 to remove warning
     wrapProgram $out/bin/bolt --set BOLT_GEM 1
   '';
+
+  passthru.tests.version = testers.testVersion {
+    package = puppet-bolt;
+    version = (import ./gemset.nix).bolt.version;
+  };
+
+  passthru.updateScript = bundlerUpdateScript "puppet-bolt";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/puppet/puppet-bolt/default.nix:26]:
```suggestion
  passthru= {
    tests.version = testers.testVersion {
      package = puppet-bolt;
      version = (import ./gemset.nix).bolt.version;
    };
    updateScript = bundlerUpdateScript "puppet-bolt";
  };
```

File: pkgs/tools/admin/puppet/puppet-bolt/default.nix [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+
+  meta = with lib; {
+    description = "Execute commands remotely over SSH and WinRM";
+    homepage = "https://github.com/puppetlabs/bolt";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ uvnikita ];
+    mainProgram = "bolt";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/puppet/puppet-bolt/default.nix:33]:
```suggestion
```
that should already be defaulted to

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/tools/admin/puppet/puppet-bolt/default.nix (hunk 2 above)
Comment (uvNikita) [reply pkgs/tools/admin/puppet/puppet-bolt/default.nix:33]: Seems like without `mainProgram = "bolt";` `testVersion` tries to call `bolt-3.24.0 --version` and fails.

Comment (SuperSandro2000) [reply pkgs/tools/admin/puppet/puppet-bolt/default.nix:34]: bundlerApp is not a real package thats why it is not working. Will be fixed universally with #182816

In the meantime you can squash all the commits.

Comment (uvNikita) [reply pkgs/tools/admin/puppet/puppet-bolt/default.nix]: Done

**Comment by uvNikita** [comment]
Should we merge this one now since https://github.com/NixOS/nixpkgs/pull/182816 was merged?

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
Not sure why this is not working.

```
/build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 1: bolt-3.24.0: command not found
```

**Comment by uvNikita** [comment]
I rebased on current master, should work now

**Pull request merged**